pyfstat/core.py

```diff
@@ -1676,9 +1695,9 @@ def _get_per_segment_det_stat(self, tstart):
                 2 * FS.F_mn.data[0][0], self.twoFX, self.BSGLSetup
             )
             d_detStat = log10_BSGL / np.log10(np.exp(1))
-        if np.isnan(d_detStat):
+        if np.isnan(np.sum(d_detStat)):
```

Comment:
I think you want instead `if np.any(np.isnan(d_detStat)):` - the result should always be the same as `[0,1,2,nan].sum()=nan` hopefully, but it makes clearer what is being tested. (This function doesn't know the caller will want to take a sum.)

Reply:
That was the first thing I implemented, and I agree with the "clearer".

However...

```
In [2]: import numpy as np

In [3]: a = np.random.random((10000, 10000))

In [4]: %timeit np.isnan(a)
51.8 ms ± 528 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [5]: %timeit np.sum(a)
37.3 ms ± 243 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [6]: %timeit np.any(np.isnan(a))
57.6 ms ± 151 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [7]: %imteit np.isnan(np.sum(a))
UsageError: Line magic function `%imteit` not found.

In [8]: %timeit np.isnan(np.sum(a))
37.3 ms ± 106 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

although for smaller arrays...
```
In [9]: a = np.random.random((100, 100))

In [10]: %timeit np.sum(a)
4.95 µs ± 33.1 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [11]: %timeit np.isnan(a)
2.19 µs ± 16.9 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [12]: %timeit np.any(np.isnan(a))
5.56 µs ± 55.1 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [13]: %timeit np.isnan(np.sum(a))
7.07 µs ± 253 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
Alternative: since `_get_per_segment_det_stat()` is only ever called from `get_semicoherent_twoF()`, just move the nan check into that one, after the sum is already taken anyway.

Reply:
Ok, I like that alternative